SimTKcommon/Simulation/src/State.cpp

```diff
@@ -312,7 +313,8 @@ StateImpl::StateImpl(const StateImpl& src)
 :   currentSystemStage(Stage::Empty)
 {
     initializeStageVersions();
-
+    stateLock = new std::mutex; //TODO: When we copy a state, do we want it to have a different stateLock? I think yes?
```

Comment:
Yes, every State object should have its own mutex. And I see that there is no std::mutex copy constructor. I still don't think you have to allocate the mutex on the heap to accomplish this though as long as there is a non-default copy constructor. Just let the mutex construct itself. Can we move a mutex?